src/domain/notification/NotificationData.ts

```diff
@@ -0,0 +1,32 @@
+
+export default class NotificationData
+{
+    #id: string;
+
+    #senderId: string;
+    #receiverId: string;
+
+    #type: string;
+    #createdAt: Date;
+
+    constructor(id: string, senderId: string, receiverId: string, type: string, createdAt = new Date())
+    {
+        this.#id = id;
+
+        this.#senderId = senderId;
+        this.#receiverId = receiverId;
+
+        this.#type = type;
+        this.#createdAt = createdAt;
```

Comment:
As discussed, the purpose of the data model is for the initial design only. We've chosen to make the code the source of truth after the design phase to prevent additional overhead.